src/node.cc

```diff
@@ -1217,7 +1218,21 @@ Local<Value> MakeCallback(Environment* env,
     }
   }
 
-  if (!env->KickNextTick(&callback_scope)) {
+  if (callback_scope.in_makecallback()) {
+    return ret;
+  }
+
+  Environment::TickInfo* tick_info = env->tick_info();
+
+  if (tick_info->length() == 0) {
+    env->isolate()->RunMicrotasks();
+  }
+
+  if (tick_info->length() == 0) {
+    tick_info->set_index(0);
```

Comment:
In `Environment::KickNextTick` and `AsyncWrap::MakeCallback` this if statement has an early return. It might not matters but at the very least it is an inconsistency between `AsyncWrap::MakeCallback` and `node::MakeCallback`.